presto-hive/src/main/java/io/prestosql/plugin/hive/BackgroundHiveSplitLoader.java

```diff
@@ -553,7 +568,8 @@ private static boolean shouldUseFileSplitsFromInputFormat(InputFormat<?, ?> inpu
             }
             if (containsEligibleTableBucket) {
                 for (LocatedFileStatus file : bucketFiles.get(partitionBucketNumber)) {
-                    splitFactory.createInternalHiveSplit(file, readBucketNumber)
+                    // TODO from deleteDeltaLocations we should read only relevant bucket files
```

Comment:
We should list files under deleteDeltaLocations and split them into buckets (using `bucket_(\\d+)`).
Then each split would cary locations of matching delete deltas only.
For this, in order to avoid code duplication, we should _replace_ `Optional<DeleteDeltaLocations>` in `HiveSplit` with `List<Path> deleteDeltaFiles`. When reading from a non-bucketed table (or bucketed v2), we would need to list the delta locations to get the files in a split. I think it's an OK approach, but would appreciate feedback.

Reply:
Hm, seems to change is needed, thanks to
https://github.com/prestosql/presto/blob/acf2c803530a0cb08704b43aea5b8a4b24609edb/presto-hive/src/main/java/io/prestosql/plugin/hive/orc/acid/DeletedRowsRegistry.java#L146-L152

> ```
> // Now filter out only the delete delta files which have same name as split's file.
> ImmutableList.Builder usefulDeleteDeltaFiles = ImmutableList.builder();
> for (LocatedFileStatus fileStatus : deleteDeltaFiles) {
>     if (fileStatus.getPath().getName().equals(splitPath.getName())) {
>         usefulDeleteDeltaFiles.add(fileStatus);
>     }
> }
> ```